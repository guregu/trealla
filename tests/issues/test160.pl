:-initialization(main).

main :-
	main1, main2, main3.

main1 :-
	NV1=nv1,
	when((nonvar(NV1) ; nonvar(NV2)), writeln(ok1)),
	writeln(here11),
	NV2=nv2,
	writeln(here12).

main2 :-
	NV1=nv1,
	when((nonvar(NV1) , nonvar(NV2)), writeln(ok2)),
	writeln(here21),
	NV2=nv2,
	writeln(here22).

main3 :-
	NV=nv,
	when(nonvar(NV), writeln(ok3)),
	writeln(here3).
