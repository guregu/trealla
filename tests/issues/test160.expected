ok1
here11
here12
here21
ok2
here22
ok3
here3
